### PR TITLE
add and use PollInterval type (taken from clock-adjust branch)

### PR DESCRIPTION
**Pull request by folkertdev**

### CODE REVIEW

File: ntp-proto/src/peer.rs [hunk 1]
```diff
@@ -152,9 +152,9 @@ impl Peer {
         let time = local_clock_time;
 
         Self {
-            last_poll_interval: 2,
-            next_poll_interval: 2,
-            remote_min_poll_interval: 2,
+            last_poll_interval: PollInterval::MIN,
+            next_poll_interval: PollInterval::MIN,
+            remote_min_poll_interval: PollInterval::MIN,
```
Comment (folkertdev) [inline ntp-proto/src/peer.rs:157]: see below for discussion on what `PollInterval::MIN` should be. I'm not sure where these defaults come from, but `MIN` is , at least at the moment, not `2`

File: ntp-proto/src/time_types.rs [hunk 2]
```diff
@@ -381,6 +382,47 @@ ntp_duration_scalar_div!(u16);
 ntp_duration_scalar_div!(u32);
 // u64 and usize deliberately excluded as they can result in overflows
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct PollInterval(i8);
```
Comment (djc) [inline ntp-proto/src/time_types.rs:386]: Could use a little documentation on what the inner value means.

File: ntp-proto/src/time_types.rs [hunk 3]
```diff
@@ -381,6 +382,47 @@ ntp_duration_scalar_div!(u16);
...
+
+impl PollInterval {
+    pub const MIN: Self = Self(4);
+    pub const MAX: Self = Self(17);
```
Comment (folkertdev) [inline ntp-proto/src/time_types.rs:390]: The situation here is interesting. The spec says (section 7.2):

    | MINPOLL   | 4     | minimum poll exponent (16 s)     |
    | MAXPOLL   | 17    | maximum poll exponent (36 h)     |

The code skeleton says

    #define MINPOLL         6       /* % minimum poll interval (64 s)*/
    #define MAXPOLL         17      /* % maximum poll interval (36.4 h) */

finally, the ntp repository 

    #define	NTP_MINPOLL	3	/* log2 min poll interval (8 s) */
    #define NTP_MINDPOLL	6	/* log2 default min poll (64 s) */
    #define NTP_MAXDPOLL	10	/* log2 default max poll (~17 m) */
    #define	NTP_MAXPOLL	17	/* log2 max poll interval (~36 h) */

So nobody agrees. Guess we follow the spec literally here?

File: ntp-proto/src/time_types.rs [hunk 4]
```diff
@@ -381,6 +382,47 @@ ntp_duration_scalar_div!(u16);
...
+
+    pub fn inc(&mut self) {
+        *self = Self(self.0 + 1).min(Self::MAX);
+    }
+
+    #[must_use]
```
Comment (djc) [inline ntp-proto/src/time_types.rs:396]: So far `inc()` and `dec()` only appear to be used in tests, whereas `plus_one()` is actually used in the library. Will that change? I would probably only have copying `inc()` and `dec()` and ditch `plus_one()`; having both seems a little weird and modifying by `&mut` isn't actually that valuable for a type that's smaller than a word.

**Review by djc — COMMENTED** [review]
I think this is a really nice change! Some nits below.

File: ntp-proto/src/time_types.rs (hunk 4 above)
Comment (folkertdev) [reply ntp-proto/src/time_types.rs:398]: I pulled this from branch, but turns out inc/dec also were not used there. I've changed the semantics of inc/dec to take/return by-value

File: ntp-proto/src/time_types.rs (hunk 3 above)
Comment (davidv1992) [reply ntp-proto/src/time_types.rs:396]: Indeed, decided to go with the main (normative) part of the spec here. We can always decide to deviate here should we want to.

File: ntp-proto/src/peer.rs (hunk 1 above)
Comment (davidv1992) [reply ntp-proto/src/peer.rs:156]: I needed some initial values, and these seemed appropriate. It seems the standard doesn't specify these

File: ntp-proto/src/time_types.rs [hunk 5]
```diff
@@ -381,6 +382,50 @@ ntp_duration_scalar_div!(u16);
...
+    pub fn inc(self) -> Self {
+        Self(self.0 + 1).min(Self::MAX)
+    }
+
+    #[must_use]
+    pub fn dec(self) -> Self {
+        Self(self.0 - 1).max(Self::MIN)
+    }
```
Comment (davidv1992) [inline ntp-proto/src/time_types.rs:406]: Not sure I agree with the change in interface here. This is going to result in a lot of code following the pattern `a=a.inc()`, which is why I originally decided on modifying inc/dec

**Pull request merged**